File: src/main/java/com/moneybags/services/BudgetsService.java
package com.moneybags.services;

import com.moneybags.domain.Budget;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Users: daniel.baker
 * Date: 3/8/2015
 * Time: 9:49 PM
 */
@Service
public interface BudgetsService {
    Budget saveBudget(Budget budget);
    List<Budget> getBudgets(Long userIdSeq);
}

File: src/main/resources/static/scripts/services/BudgetsService.js
/**
 * Created by daniel.baker on 3/1/2015.
 */
(function(){
    var budgets = function($http) {
        var getBudgets = function(userIdSeq) {
            return $http.post("../getBudgets", userIdSeq)
                .then(function(response) {
                    return response.data;
                });
        };
        var saveBudget = function(budget) {
            return $http.post("../saveBudget", budget)
                .then(function(response) {
                    return response.data;
                });
        };

        return {
            getBudgets: getBudgets,
            saveBudget: saveBudget
        };
    };
    var module = angular.module("moneybags");
    module.factory("budgets", budgets);
}());